internal/controller/instaslice_controller.go

```diff
@@ -275,17 +315,23 @@ func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 				}
 			}
 		}
+		gpuOperatorPodOk := false
+		var err error
 		for _, instaslice := range instasliceList.Items {
 			for podUuid, allocations := range instaslice.Spec.Allocations {
 				if allocations.Allocationstatus == inferencev1alpha1.AllocationStatusCreated && allocations.PodUUID == string(pod.UID) {
+					gpuOperatorPodOk, err = r.isPatternPodRunningAndHealthy(ctx, "nvidia-device-plugin-daemonset", "gpu-operator")
+					if err != nil {
+						log.FromContext(ctx).Info("gpu operator pod not found")
```

Comment:
err type should be checked for not found, and if it is a different error then handle it gracefully.

Reply:
Done, thanks